### ISSUE TITLE
Coordinates from NEOS are in some ecliptic reference frame

### ISSUE THREAD

**Issue by Juanlu001** [poliastro/poliastro#222]
...whereas planetary ephemerides are in ICRS. Unfortunately, Astropy 2.0.1 does not support velocities for ecliptic coordinate frames:

https://github.com/astropy/astropy/issues/6494

We should, at the very least, document this issue because it can potentially produce misguided plots (as we have been doing all along...)

**Comment by Juanlu001** [comment]
Of course the right thing to do is to contribute velocity transformations for `Heliocentric/BarycentricTrueEcliptic` coordinates in Astropy.

**Comment by Juanlu001** [comment]
The situation is analyzed in depth in the wiki:

https://github.com/poliastro/poliastro/wiki/Reference-frames

**Comment by Juanlu001** [comment]
Correction: Astropy does support velocities in those frames, but another calling convention must be used:

```
HeliocentricTrueEcliptic(
    x=florence.r[0], y=florence.r[1], z=florence.r[2],
    d_x=florence.v[0], d_y=florence.v[1], d_z=florence.v[2],
    representation=CartesianRepresentation,
    differential_cls=CartesianDifferential
)
```

**Comment by Juanlu001** [comment]
Asked the Astropy folks about our concerns: https://github.com/astropy/astropy/pull/3749#issuecomment-326320598

**Comment by Juanlu001** [comment]
Differences in plotting:

![differences](https://user-images.githubusercontent.com/316517/29931181-cc6ab15c-8e70-11e7-9d09-a5b3286aa4ec.png)

(code WIP)

**Comment by Juanlu001** [comment]
Progress: Although it was stated in the wiki that the results from Horizons and the Small Object Browser are not exactly the same, in fact all the differences stay within the declared uncertainties (1 sigma), so we can consider them equal in some sense :slightly_smiling_face: 

And on the other hand, this is the information Horizons gives about the reference frame:

```
Coordinate system description:

  Ecliptic and Mean Equinox of Reference Epoch

    Reference epoch: J2000.0
    XY-plane: plane of the Earth's orbit at the reference epoch
              Note: obliquity of 84381.448 arcseconds wrt ICRF equator (IAU76)
    X-axis  : out along ascending node of instantaneous plane of the Earth's
              orbit and the Earth's mean equator at the reference epoch
    Z-axis  : perpendicular to the xy-plane in the directional (+ or -) sense
              of Earth's north pole at the reference epoch.
```

With respect to the obliquity, this is the result of using Astropy functions:

```
In [7]: from astropy import _erfa as erfa

In [8]: from astropy.time import Time 

In [9]: equinox = Time("J2000")

In [11]: from astropy.coordinates.builtin_frames.utils import get_jd12

In [12]: jd1, jd2 = get_jd12(equinox, 'tt')

In [13]: from astropy import units as u

In [14]: obl = erfa.obl06(jd1, jd2)*u.radian

In [15]: obl
Out[15]: <Quantity 0.40909260059596453 rad>

In [20]: obl.to(u.arcsec)
Out[20]: <Quantity 84381.4059990474 arcsec>
```

which is slightly different from the announced value of 84381.448 arcseconds. This is because Horizons uses the ICRF equator of IAU76!

```
In [23]: obl80 = erfa.obl80(jd1, jd2) * u.radian

In [24]: obl80
Out[24]: <Quantity 0.40909280421771277 rad>

In [25]: obl80.to(u.arcsecond)
Out[25]: <Quantity 84381.44799904784 arcsec>
```

So we need a *mean* ecliptic of IAU76, which is not present in Astropy.

**Comment by Juanlu001** [comment]
Bad news: this did not lead to better results, and in fact the errors were very similar. Furthermore, after https://github.com/astropy/astropy/pull/6508 was submitted, I am not sure whether the ecliptic coordinate frames present in Astropy should be renamed to `Mean` instead of `True`.

On the other hand, I did not find any obvious relationship between the ground truth result (the one that Horizons gives) and the conversion we get. The difference in norms is of the same order of magnitude than the norm of the position vector of the Sun with respect to the Solar System Barycenter, but it can be just coincidence.

I officially finished all the black box approaches to this issue. Next steps are:

1. Reading the source code of SPICE.
2. Contacting NASA.

But that will be for another week.

**Comment by Juanlu001** [comment]
We did it!

https://github.com/astropy/astropy/pull/6508#issuecomment-327004169

**Comment by Juanlu001** [comment]
There's some discussion going on, but I intend to fix this for 0.8.0.

**Comment by Juanlu001** [comment]
No way: I want this fixed for 0.7.0 :smile:
